infrastructure/resource_group_level/modules/cosmosdb.bicep

```diff
@@ -0,0 +1,564 @@
+@allowed(['Provisioned Throughput', 'Serverless'])
+param capacityMode string
+
+var cosmosDbName = 'tfs-${environment}-mongo'
+var privateEndpointName = 'tfs-${environment}-mongo'
+
+
+// TODO:FN-416 Work out what these misc. IPs with access are.
+// Misc IPs
+var miscIps =  [
+  '86.130.182.52'
+  '81.154.165.161'
+  '86.139.72.145'
+] 
+
+// On activating "Allow access from Azure Portal":
+// https://github.com/Azure/azure-cli/issues/7495 ->
+// https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-configure-firewall#allow-requests-from-the-azure-portal
+// As mentioned above, adding the following is equivalent to checking "Allow access from Azure Portal" "Accept connections from within public Azure datacenters" in the portal.
+var azurePortalIps = [
+  '104.42.195.92'
+  '40.76.54.131'
+  '52.176.6.30'
+  '52.169.50.45'
+  '52.187.184.26'
+]
+
+// As mentioned above, adding the following is equivalent to checking "Accept connections from within public Azure datacenters" in the portal.
+// We need this for Azure functions.
+var acceptPublicAzureDatacentersIp = ['0.0.0.0']
+
+var allowedIps = concat(onPremiseNetworkIps, miscIps, azurePortalIps, acceptPublicAzureDatacentersIp)
+
+var capabilities = capacityMode == 'Provisioned Throughput' ? [
+  {
+    name: 'EnableMongo'
+  }
+] : [
+  {
+    name: 'EnableMongo'
+  }
+  {
+    name: 'DisableRateLimitingResponses'
+  }
+  {
+    name: 'EnableServerless'
+  }  
+]
+
+
+resource cosmosDb 'Microsoft.DocumentDB/databaseAccounts@2023-04-15' = {
+  name: cosmosDbName
+  location: location
+  tags: {
+    Environment: 'Preproduction'
+  }
+  kind: 'MongoDB'
+  identity: {
+    type: 'None'
+  }
+  properties: {
+    publicNetworkAccess: 'Enabled'
+    enableAutomaticFailover: false
+    enableMultipleWriteLocations: false
+    isVirtualNetworkFilterEnabled: true
+    virtualNetworkRules: [
+      {
+        id: appServicePlanEgressSubnetId
+        ignoreMissingVNetServiceEndpoint: false
+      }
+    ]
+    disableKeyBasedMetadataWriteAccess: false
+    enableFreeTier: false
+    enableAnalyticalStorage: false
+    analyticalStorageConfiguration: {}
+    createMode: 'Default'
+    databaseAccountOfferType: 'Standard'
+    defaultIdentity: 'FirstPartyIdentity'
+    networkAclBypass: 'None'
+    disableLocalAuth: false
+    enablePartitionMerge: false
+    minimalTlsVersion: 'Tls12'
+    consistencyPolicy: {
+      defaultConsistencyLevel: 'Session'
+      maxIntervalInSeconds: 5
+      maxStalenessPrefix: 100
+    }
+    apiProperties: {
+      serverVersion: '4.2'
+    }
+    locations: [
+      {
+        locationName: location
+        failoverPriority: 0
+        isZoneRedundant: false
+      }
+    ]
+    cors: []
+    capabilities: capabilities
+    ipRules: [for ip in allowedIps: {
+      ipAddressOrRange: ip
+    }]
+    backupPolicy: {
+      type: 'Continuous'
+      continuousModeProperties: {
+        tier: 'Continuous30Days'
```

Comment:
Note that this is paid for. I think we should probably use the free 7 days backup for all environments apart from live.
Agreed? If so, I'll add it in.

Reply:
yup, good shout